src/serve_style.js

```diff
@@ -42,33 +42,76 @@ export const serve_style = {
       }
       // mapbox-gl-js viewer cannot handle sprite urls with query
       if (styleJSON_.sprite) {
-        styleJSON_.sprite = fixUrl(req, styleJSON_.sprite, item.publicUrl);
+        if (Array.isArray(styleJSON_.sprite)) {
+          styleJSON_.sprite.forEach((spriteItem) => {
+            spriteItem.url = fixUrl(req, spriteItem.url, item.publicUrl);
+          });
+        } else {
+          styleJSON_.sprite = fixUrl(req, styleJSON_.sprite, item.publicUrl);
+        }
       }
       if (styleJSON_.glyphs) {
         styleJSON_.glyphs = fixUrl(req, styleJSON_.glyphs, item.publicUrl);
       }
       return res.send(styleJSON_);
     });
 
-    app.get('/:id/sprite:scale(@[23]x)?.:format([\\w]+)', (req, res, next) => {
-      const item = repo[req.params.id];
-      if (!item || !item.spritePath) {
-        return res.sendStatus(404);
-      }
-      const scale = req.params.scale;
-      const format = req.params.format;
-      const filename = `${item.spritePath + (scale || '')}.${format}`;
-      return fs.readFile(filename, (err, data) => {
-        if (err) {
-          console.log('Sprite load error:', filename);
+    app.get(
+      '/:id/sprite(/:spriteID)?:scale(@[23]x)?.:format([\\w]+)',
+      (req, res, next) => {
+        const spriteID = req.params.spriteID || 'default';
+        const scale = req.params.scale || '';
+        const format = req.params.format;
+        const item = repo[req.params.id];
+
+        let spritePath;
+        if (item && item.spritePaths) {
+          for (const sprite of item.spritePaths) {
+            if (sprite.id === spriteID) {
+              spritePath = sprite.path;
+            }
+          }
+          if (!spritePath) {
+            return res.sendStatus(404);
+          }
+        } else {
           return res.sendStatus(404);
+        }
+
+        let spriteScale;
+        const allowedScales = ['', '@2x', '@3x'];
```

Comment:
To be honest I only included this to make the CodeQL ci happy. It didn't like user supplied values getting used in path later.

Though it could make sense to move it use it in other places, though I'm not sure how many places use the same scales